package protocol.effect.plexus;
import "common.proto";

message Plexus
{
  repeated TextPath text_paths = 1;
  repeated NoiseEffector noise_effectors = 2;
}

message TextPath
{
  optional string text = 1;
}

message Displacement
{
	optional protocol.common.FloatAnim x = 1;
	optional protocol.common.FloatAnim y = 2;
	optional protocol.common.FloatAnim z = 3;
}

message NoiseEffector
{
  enum ApplyTo {
    Position = 1;
    Scale = 2;
  };
  optional ApplyTo apply_to = 1 [default = Position];
  optional Displacement displacement = 2;
}
